### PR TITLE
Feature/grader UI

### PR DESCRIPTION
**Pull request by kennybc**
# Checklist

- [ ] I have performed a self-review of my code
- [ ] I have reached out to another developer to review my code
- [ ] New and existing unit tests pass locally with my changes
- [ ] All Lint and CI checks are passing

### CODE REVIEW

**Review by alexangione419 — COMMENTED** [review]
backend looks good, want someone to look at front end still

File: backend/internal/handlers/file_tree/service.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"github.com/CamPlume1/khoury-classroom/internal/github"
+	"github.com/CamPlume1/khoury-classroom/internal/storage"
+)
+
+type FileTreeService struct {
+	store           storage.Storage
+	githubappclient github.GitHubAppClient
+}
+
+func newFileService(store storage.Storage, githubappclient github.GitHubAppClient) *FileTreeService {
```
Comment (CamPlume1) [inline backend/internal/handlers/file_tree/service.go:13]: Capitalize N if you want it to export

File: backend/internal/github/appclient/appclient.go [hunk 2]
```diff
@@ -56,7 +60,7 @@ func (api *AppAPI) GetJWT(ctx context.Context) (string, error) {
 	return token.AccessToken, nil
 }
 
-func (api *AppAPI) ListInstallations(ctx context.Context) ([]*github.Installation, error) {
+func (api *AppAPI) GetClientWithJWTAuth(ctx context.Context) (*github.Client, error) {
```
Comment (CamPlume1) [inline backend/internal/github/appclient/appclient.go:63]: You shouldn't be returning a github client, this should be a struct field.

Think of it as a private field in Java with a wrapper class.

Okay as a private method if you rename with lowercase G

File: backend/internal/storage/postgres/student_assignments.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	rows, err := db.connPool.Query(ctx,
+		"SELECT * FROM student_assignments WHERE assignment_id = (SELECT id FROM assignments WHERE classroom_id = $1 ORDER BY name ASC OFFSET $2 LIMIT 1) ORDER BY student_gh_username ASC",
+		classroomID, localAssignmentID-1)
+	// -1 for 1-indexing
+
+	var studentAssignment []models.StudentAssignment
+
+	if err != nil {
+		fmt.Println("Error in query")
+		return studentAssignment, err
```
Comment (alexangione419) [inline backend/internal/storage/postgres/student_assignments.go:21]: should prolly be nil

File: frontend/src/pages/Dashboard/index.tsx [hunk 4]
```diff
@@ -55,104 +59,125 @@ const Dashboard: React.FC = () => {
...
-            } catch (error: unknown) {
-                console.error('Error making API call:', error);
+  useEffect(() => {
+    const fetchAssignments = async (semester: ISemester) => {
+      try {
+        if (semester.classroom_id) {
+          const base_url: string = import.meta.env
+            .VITE_PUBLIC_API_DOMAIN as string;
+          const result = await fetch(
+            `${base_url}/assignments/${semester.classroom_id}`,
```
Comment (alexangione419) [inline frontend/src/pages/Dashboard/index.tsx:69]: needs to be updated to the new route for getting all assignments with a certain classroom id

**Review by alexangione419 — CHANGES_REQUESTED** [review]

**Review by nandini-ghosh — COMMENTED** [review]
Frontend looks good, contexts make sense to me

**Review by CamPlume1 — APPROVED** [review]

File: backend/internal/handlers/file_tree/service.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package file_tree
+
+import (
+	"github.com/CamPlume1/khoury-classroom/internal/github"
+	"github.com/CamPlume1/khoury-classroom/internal/storage"
+)
+
+type FileTreeService struct {
```
Comment (ntietje1) [inline backend/internal/handlers/file_tree/service.go:8]: Does this need to be a separate service? It seems like this is still a "github" related feature

File: backend/internal/handlers/semesters/assignments/assignments.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+	"github.com/CamPlume1/khoury-classroom/internal/errs"
+	"github.com/CamPlume1/khoury-classroom/internal/models"
+	"github.com/gofiber/fiber/v2"
+)
+
+func (s *AssignmentsService) GetAssignmentsInSemester(c *fiber.Ctx) error {
+	classroomID, err := strconv.ParseInt(c.Params("classroomID"), 10, 64)
+	if err != nil {
+		return err
```
Comment (ntietje1) [inline backend/internal/handlers/semesters/assignments/assignments.go:16]: What status code will this return? I think we should explicitly set them in handlers

File: backend/internal/handlers/semesters/assignments/assignments.go [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+	if err != nil {
+		fmt.Println("error in service func", err)
+		return err
+	}
+
+	return c.Status(http.StatusOK).JSON(assignments)
+}
+
+func (s *AssignmentsService) GetAssignment(c *fiber.Ctx) error {
+	classroomID, _ := strconv.ParseInt(c.Params("classroomID"), 10, 64)
```
Comment (ntietje1) [inline backend/internal/handlers/semesters/assignments/assignments.go:29]: you don't handle the error case here

File: backend/internal/handlers/semesters/assignments/student_assignments/student_assignments.go [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+func (s *StudentAssignmentService) GetStudentAssignments(c *fiber.Ctx) error {
+	classroomID, _ := strconv.ParseInt(c.Params("classroomID"), 10, 64)
```
Comment (ntietje1) [inline backend/internal/handlers/semesters/assignments/student_assignments/student_assignments.go:12]: handle error case

File: backend/internal/handlers/semesters/assignments/student_assignments/student_assignments.go [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+	if err != nil {
+		fmt.Println("error in service func")
+		return err
+	}
+
+	return c.Status(http.StatusOK).JSON(studentAssignment)
+}
+
+func (s *StudentAssignmentService) GetStudentAssignment(c *fiber.Ctx) error {
+	classroomID, _ := strconv.ParseInt(c.Params("classroomID"), 10, 64)
```
Comment (ntietje1) [inline backend/internal/handlers/semesters/assignments/student_assignments/student_assignments.go:25]: error case

File: backend/internal/storage/postgres/assignments.go [hunk 10]
```diff
@@ -79,3 +79,17 @@ func (db *DB) GetAssignmentIDs(ctx context.Context) ([]models.Assignment_Classro
 	return pgx.CollectRows(rows, pgx.RowToStructByName[models.Assignment_Classroom_ID])
 
 }
+
+func (db *DB) GetAssignment(ctx context.Context, classroomID int64, localAssignmentID int64) (models.Assignment, error) {
+	rows, err := db.connPool.Query(ctx,
+		"SELECT * FROM assignments WHERE classroom_id = $1 ORDER BY inserted_date ASC OFFSET $2 LIMIT 1",
+		classroomID,
+		localAssignmentID-1)
```
Comment (ntietje1) [inline backend/internal/storage/postgres/assignments.go:87]: I'd like at least a comment explaining why we have a magic "-1" here

**Review by ntietje1 — CHANGES_REQUESTED** [review]

**Review by ntietje1 — APPROVED** [review]
66b3e5d0c2ab246786ca1d5e_86

**Pull request merged**